include/clock_control.h

```diff
@@ -36,10 +51,20 @@ typedef int (*clock_control_get)(struct device *dev,
 				 clock_control_subsys_t sys,
 				 u32_t *rate);
 
+typedef int (*clock_async_control)(struct device *dev,
+				   clock_control_subsys_t sys,
+				   struct clock_control_user *user,
+				   u32_t when);
+
+typedef enum clock_control_status (*clock_control_get_status)(
+							struct device *dev);
+
 struct clock_control_driver_api {
-	clock_control		on;
-	clock_control		off;
-	clock_control_get	get_rate;
+	clock_control			on;
+	clock_control			off;
+	clock_async_control		async_on;
+	clock_control_get		get_rate;
```

Comment:
clock_control_rate_get rate_get; /* if this as a API change is permitted, to follow a consistent naming */

Reply:
hum, why changing the order? get_rate seems totally fine here.

Reply:
Kernel APIs and most functions in Zephyr have the verb as the last word, makes sense to be consistent.

Reply:
No need to be strict about it. Actually many other device API do not follow this.
It would be valid if other functions from the same API were reversed.

Reply:
I agree with @tbursztyka here. I see no reason to change it. Also @cvinayak although kernel primitives use the `<object>_<predicate>` format, this is quite the opposite for the drivers and subsystems, and this is a driver